### PR TITLE
Supply folder parameter during creation so that a subsequent move is not needed

### PR DESCRIPTION
**Pull request by MikeTschudi**
The REST API's [createService](https://developers.arcgis.com/rest/users-groups-and-items/create-service.htm) allows a folder to be specified, so instead of creating a service and then moving it to the desired folder, we simply create it in the desired folder.

Originally didn't think that this was possible ([Added create feature service #307](https://github.com/Esri/arcgis-rest-js/pull/307)), but @chris-fox suggested saving the extra round trip to the server.

### CODE REVIEW

File: __reinstallNodeModules_rest.bat [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+call rmdir/q/s packages\arcgis-rest-geocoding\node_modules
+call rmdir/q/s packages\arcgis-rest-portal\node_modules
+call rmdir/q/s packages\arcgis-rest-request\node_modules
+call rmdir/q/s packages\arcgis-rest-routing\node_modules
+call rmdir/q/s packages\arcgis-rest-service-admin\node_modules
+
+call npm install
+
+call npm run test:all
+
```
Comment (dbouwman) [inline __reinstallNodeModules_rest.bat:13]: I think you could achieve this in a cross-platform manner using `rimraf` in a npm script, similar to how the `docs:serve` script does it. Alternatively you could also run [windows bash](https://www.howtogeek.com/249966/how-to-install-and-use-the-linux-bash-shell-on-windows-10/)

File: __createTag.bat [hunk 2]
```diff
@@ -0,0 +1,4 @@
+for %%F in ("%cd%") do set "folder=%%~nxF"
```
Comment (dbouwman) [inline __createTag.bat:1]: I'm not really clear what this script does - when we create releases and/or tags, normal git commands seem to work... do those not work for you? or is this for some other purpose?

Comment (MikeTschudi) [reply __createTag.bat:1]: Thanks, @dbouwman; these are errors--I have a different .gitignore file that excludes tool files prefixed with "__".  I'll get rid of them.

Comment (dbouwman) [reply __createTag.bat:1]: sounds good! thanks!

**Comment by tomwayson** [comment]
Thanks for reviewing this @dbouwman 

Let's hold on merging this until I get https://github.com/Esri/arcgis-rest-js/pull/703 merged and released

**Comment by tomwayson** [comment]
Merge and release at will https://github.com/Esri/arcgis-rest-js/pull/703#issuecomment-627346705

**Pull request merged**